ql/src/java/org/apache/hadoop/hive/ql/exec/MoveTask.java

```diff
@@ -758,8 +759,8 @@ private DataContainer handleDynParts(Hive db, Table table, LoadTableDesc tbd,
       }
       LOG.info("Loading partition " + entry.getKey());
     }
-    console.printInfo("\t Time taken for adding to write entity : " +
-        (System.currentTimeMillis() - startTime)/1000.0 + " seconds");
+    console.printInfo(String.format("Time taken for adding to write entity:\t %.3f seconds",
+        (System.currentTimeMillis() - startTime) / 1000.0));
```

Comment:
this entire class does this not something new, but still I will take the opportunity here to say. ``System.currentTimeMillis()`` isn't the best thing to do when calculating elaspsed time, it is subject to wall clock changes, Should use ``nanoTime()`` always, or Time.monotonicNow()

https://github.com/apache/hadoop/blob/31b8b4a8bd5602e04e780776acf878dacda01656/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Time.java#L49-L69

Reply:
right, we're doing this currentTimeMillis thing all the time :D
I agree, let me fix this as I'm touching this part now